controllers/amazoncloudwatchagent_controller.go

```diff
@@ -40,6 +43,80 @@ type Params struct {
 	Config   config.Config
 }
 
+func (r *AmazonCloudWatchAgentReconciler) findCloudWatchAgentOwnedObjects(ctx context.Context, owner v1alpha1.AmazonCloudWatchAgent) (map[types.UID]client.Object, error) {
```

Comment:
Can you set up a unit test for this function with a sample `v1alpha1.AmazonCloudWatchAgent` instance and see if this properly detects the created resources?

Reply:
There are no unit-test available for controllers, they are tested via integration tests